test/lifecycle_test.dart

```diff
@@ -139,7 +140,78 @@ void main() {
       ]));
     });
```

Comment:
There should be a lifecycle tests that asserts that `getChildContext` is not called when `childContextKeys` is empty.

Reply:
Tests incoming. Good call here.

Reply:
See 
282914b